Replace faker package with @faker-js/faker

EdwardSalter opened the pull request:
Replaces the _faker_ package with _@faker-js/faker_.

NOTE: Due to some restructuring in the _@faker-js/faker_ package, this upgrade should be deemed a breaking change since schemas making use of deprecated APIs will no longer function (i.e. `x-faker: random.number` needs to be replaced with `x-faker: datatype.number`).

Fixes #62

EdwardSalter commented:
To mitigate the breaking changes, we could potentially downgrade from _@faker-js/faker@7.3.0_ to _@faker-js/faker@5.5.3_ which is identical code to the original _faker_ package.

Alternatively, we could add some minimal code to alias the original methods that have been renamed in v6+to continue supporting those for a period of time ensuring that we write deprecation messages to the logger.

EdwardSalter commented:
Since v6 of _@faker-js/faker_, it is possible to create your own instance of the Faker class since they export the constructor, not just the singleton instance. It might be nice for a way for developers to optionally inject their own faker instance into the system instead of relying on the singleton provided by the package? I am just not sure how this could be handled since the ResponseGenerator is a static class at the minute also.

jormaechea commented:
@EdwardSalter hi, sorry for the delay..

I think that in order to avoid supporting old stuff and to move forward, best option is to just make this a major version with it's breaking change.
This can be a nice opportunity to also drop support for old node versions (10 and 12) and add support for newer versions (16 and 18). And also migrate the Docker image to LTS (currently using node 10 😅).

About your last comment, I think it's a nice feature, but that definitely deserves it's own issue to discuss it.

jormaechea commented:
Just created the [roadmap/v2](https://github.com/jormaechea/open-api-mocker/tree/roadmap/v2) branch to work without breaking `master`.

Can you change the PR target to that branch?

Thanks!

EdwardSalter commented:
> Just created the [roadmap/v2](https://github.com/jormaechea/open-api-mocker/tree/roadmap/v2) branch to work without breaking `master`.
> 
> Can you change the PR target to that branch?
> 
> Thanks!

This should all be done. I rebased my branch, fixed the merge conflicts then force pushed. I've double checked the Files Changed tab and all the changes look relevant still.
